frontends/web/src/components/rates/rates.tsx

```diff
@@ -143,13 +138,19 @@ function Conversion({
   noAction,
   children,
 }: PropsWithChildren<Props>): JSX.Element | null {
-  if (!rates) {
-    return null;
-  }
+
   const coin = amount.unit;
-  let formattedValue = '';
-  if (rates[coin]) {
-    formattedValue = formatCurrency(rates[coin][active] * Number(amount.amount), active);
+  let formattedValue = '---';
+
+  if (amount.conversions) {
+    if (amount.conversions[active] !== '')
+      formattedValue = amount.conversions[active];
+  } else {
+    if (rates && rates[coin]) {
+      // TODO This solution is not the best: it is better to have the backend calculating the value,
+      // with better precision, using coins/convertToFiat endpoint
+      formattedValue = formatCurrency(rates[coin][active] * Number(amount.amount), active);
```

Comment:
> > FiatConversion component modified to use high-precision backend-calculated fiat amount
> > conversions when available, instead of doing the math in the frontend with lower precision
> 
> Is this the only reason? because the backend doesn't use high precision. it uses floating point arith, see
> 
> https://github.com/digitalbitbox/bitbox-wallet-app/pull/1724/files#diff-7a00117689d4febac66fe55f84fd7cb00c99873d547c480c09d1416b57e33647R49
> 
> it seems the actual reason is because the historical rates are only available in the backend and not the frontend.

I think you are right, both frontend and backend use 64bit floating point for the math, and we only have historical rates in the backend. I talked with @thisconnect when I started working on this issue and I understood that it was better to avoid coin conversions in frontend, but maybe I remember wrong and the reason was not the precision?

Reply:
BBApp usually never calculates in the frontend and I assumed backend does more precise calculation sorry :) but now it could right?

Reply:
Oh, ok! In that case I would just remove the comment at https://github.com/Beerosagos/bitbox-wallet-app/blob/transaction-detail/frontends/web/src/components/rates/rates.tsx#L150-L151

Ok?

Reply:
no strong opinion, but the comment is still valid or not?

Reply:
I think that, until the precisions in the backend and in frontend are the same, this comment loses its sense.